### PR TITLE
Add `is_prime(char)` check to GF(char, k)

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves https://github.com/Nemocas/Nemo.jl/issues/1493 by adding an analogous check as for the modulus of a polynomial in `ZZ/nZZ` in that constructor.

cc @mjrodgers

### CODE REVIEW

File: test/flint/fq-test.jl [hunk 1]
```diff
@@ -258,3 +258,8 @@ end
    R, x = finite_field(ZZ(19), 3, "x")
    @test R([1, 0, 1]) == x^2 + 1
 end
+
+@testset "Nemo.jl#1493" begin
+   @test_throws DomainError GF(ZZ(4), 2)
+   @test_throws DomainError finite_field(ZZ(6), 2, "x")
+end
```
Comment (thofma) [inline test/flint/fq-test.jl:265]: Needs a newline.

File: test/flint/fq_nmod-test.jl [hunk 2]
```diff
@@ -270,3 +270,8 @@ end
    R, x = finite_field(19, 3, "x")
    @test R([1, 0, 1]) == x^2 + 1
 end
+
+@testset "Nemo.jl#1493" begin
+   @test_throws DomainError GF(4, 2)
+   @test_throws DomainError finite_field(6, 2, "x")
+end
```
Comment (thofma) [inline test/flint/fq_nmod-test.jl:277]: Needs a newline.

**Review by thofma — CHANGES_REQUESTED** [review]
Maybe make it a `ArgumentError`? That would be consistent with what `@req` does.

**Comment by lgoettgens** [comment]
> Maybe make it a `ArgumentError`? That would be consistent with what `@req` does.

A DomainError is consistent with the other constructors. Furthermore, Nemo uses DomainErrors a lot, so I would rather change that in a future PR (maybe once `@req` is moved to AA).

**Pull request merged**